Rework internal procedures on writing documentation (#306)
ferrocene/ferrocene: ferrocene/doc/internal-procedures/src/docs/sphinx-extensions.rst

      .. cli::option: --target <name>

         Used to specify the target triple.

      .. cli::option: --help

         Show the help message.

Linking to CLI options
~~~~~~~~~~~~~~~~~~~~~~

It's possible to link to CLI options (even defined in other documents) with the
``:cli:option:`` role. Options are identified with both the program name and
the option name, separated by a space. It works like a normal link role, so you
can either put the option name as the role value, or arbitrary text (with the
option name between ``<>``).

.. code-block:: rst

   :cli:option:`rustc --help`
   :cli:option:`the target flag <rustc --target <name>>`

``ferrocene_toctrees`` extension
--------------------------------

This extension tweaks Sphinx's table of contents support to better suit our
style of documents. It must be enabled in all of our documents.

Continuous numbering of sections
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

By default Sphinx numbers pages within a ``toctree`` directive without
considering other directives. This means adding multiple ``toctree``\ s (for
example to group some pages together) would reset the numbering, potentially
having multiple pages numbered "1".

> The extension patches Sphinx to fix this, and ensure the page numbers

Reviewer: ```suggestion
The extension patches Sphinx to fix this, and ensures that the page numbers
```

constantly increase even when multiple ``toctree`` directives are present.

Appendices support
~~~~~~~~~~~~~~~~~~

The extension adds a new ``appendices`` directive, with the same syntax and

> functionality as the ``toctree`` directive. The only difference is that page

Reviewer: ```suggestion
functionality as the ``toctree`` directive. The only difference is that pages
```

  target used for bare metal testing; it should be omitted if no special target
  was used
* ``remote_testing`` (optional): whether the tests were executed on CI or on a
  remote machine/emulator; its presence without a value means ``true``, while
  its absence means ``false``

.. code-block:: rst

   .. render-outcomes-template:: templates/tests.jinja2
      :host: x86_64-unknown-linux-gnu
      :target: x86_64-unknown-linux-gnu

   .. render-outcomes-template:: templates/tests.jinja2
      :host: x86_64-unknown-linux-gnu
      :target: aarch64-unknown-none
      :bare_metal_test_target: aarch64-unknown-ferrocenecoretest
      :remote_testing:

Rendering a summary of all test outcome pages
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The ``render-summary`` directive renders a table with all
``render-outcomes-template`` invocations present in the rest of the document.
It accepts no arguments nor options.

``ferrocene_relnotes`` extension
--------------------------------

This extension is specific to the :doc:`release-notes:index`.

Marking releases as upcoming
~~~~~~~~~~~~~~~~~~~~~~~~~~~~

The extension allows marking a page as upcoming by adding

> ``:upcoming-release:`` *before* any content of the page (execept for comments).

Reviewer: ```suggestion
``:upcoming-release:`` *before* any content of the page (except for comments).
```

This adds an "upcoming" badge next to the title, and a caution message just
below the title.

Including Rust changelogs
~~~~~~~~~~~~~~~~~~~~~~~~~

It is possible to inject the Rust changelog for a range of versions with the

> ``rust-changelog``. The directive requires the ``:from:`` and ``:to:`` options

Reviewer: ```suggestion
``rust-changelog`` directive. This directive requires the ``:from:`` and ``:to:`` options
```
